### ISSUE TITLE
[Bug]: Clicking on the right most 20px of the search box does not activate the search box

### ISSUE THREAD

**Issue by vaishngupta23** [microsoft/fluentui#32014]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 36.42 GB / 63.95 GB
  Browsers:
    Edge: Chromium (126.0.2592.87)
    Internet Explorer: 11.0.22621.3527
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://stackblitz.com/run?file=src%2Fexample.tsx

## Actual Behavior
Clicking on the right most 20px of the search box does not activate the search box 

## Expected Behavior
Search box should be activated by clicking anywhere on the search box

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by sopranopillow** [comment]
While the repro link doesn't work, I do see the issue in our docsite

**Comment by sopranopillow** [comment]
tldr we need to do the same as #30825
